src/app/components/curriculum-creator-page/curriculum-creator-page.component.spec.ts

```diff
@@ -51,19 +59,556 @@ describe('CurriculumCreatorPageComponent', () => {
         FormsModule,
         MatTooltipModule,
         ToastrModule.forRoot(),
-        HttpClientTestingModule
+        HttpClientTestingModule,
+        BrowserAnimationsModule
       ]
     })
     .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(CurriculumCreatorPageComponent);
+    toastrService = TestBed.get(ToastrService);
+    curriculumFetcherService = TestBed.get(CurriculumFetcherService);
     component = fixture.componentInstance;
+    curriculum = new Curriculum(1,"Java",[]);
+    component.activeCurriculum = null;
+    curriculumModule = new CurriculumModule(1,1,1,null);
+    curriculumModule2 = new CurriculumModule(2,2,2,null);
+    curriculumModules = [];
+    mockEvent = {currentIndex: 0, previousIndex: 0}
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
```

Comment:
Lots of tests, which is nice to have. However, there is once again zero comments explaining what is going on. This is important.